### ISSUE TITLE
test_LEDA_IO fails when LEDA is available

### ISSUE THREAD

**Issue by lrineau** [CGAL/cgal#1788]
https://cgal.geometryfactory.com/CGAL/testsuite/results-4.10-I-84.shtml#Installation

The red cells correspond to ArchLinux platforms with LEDA installed.

https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-4.10-I-84/Installation/TestReport_lrineau_ArchLinux.gz
```
- ProgramOutput.test_LEDA_IO.ArchLinux
------------------------------------------------------------------
CGAL::Random()::get_seed() = 1482726601
./cgal_test_with_cmake: line 70:  7942 Killed                  ./test_LEDA_IO
```
The program is killed by our testing scripts, because its run lasts too long.

**Comment by afabri** [comment]
@akobel Can you have a look in a debugger what happens with the testprogram. Note that the non-free edition of Leda works. The platform is called x86-64_Darwin-13.0_Apple-clang-5.1_Release-LEDA-without-GMP.

**Comment by akobel** [comment]
Not tested within the Docker build (since I still have no clue how to do that).
Locally, I see that `leda::operator>>(std::istream&, leda::rational&)` enters an infinite loop on the first input, probably desperately waiting for some whitespace. Not sure about the exact requirements of valid inputs; from non-exhaustive tests, and lacking any reasonable source of documentation, I have to assume that the parser shipped with leda-free-6.3 only handles input of the form
```
<numerator>/<denominator><whitespace><random ignored stuff>
```
Both the slash and the whitespace (different from EOF) seem to be non-negotiable.

Other honorable mentions of strange behavior:
```
input: "0/1"
input.fail() is set!
```
(no whitespace after 1)
```
input: "0"

LEDA ERROR HANDLER
	rational input: syntax error. (_rational.cpp:557)

#00 at 0x9b7341d7 in leda::trace_stack(char**)
#01 at 0x9b734372 in leda::trace_stack(std::basic_ostream<char, std::char_traits<char> >&)
#02 at 0x9b5af5b7 in leda::default_error_handler(int, char const*)
#03 at 0x9b7c10ad in leda::operator>>(std::basic_istream<char, std::char_traits<char> >&, leda::rational&)
#04 at 0x00401add in main
#05 at 0x9a549291 in __libc_start_main
#06 at 0x00401f0a in _start
```

No, I don't really believe that LEDA does not offer anything more robust - but it's what I observe. I don't have access at the sources to double-check, though.

**Comment by akobel** [comment]
That being said, the stream input works as expected on certain, reasonable inputs. Hence I'm inclined to trust that the LEDA-CGAL-interface is working fine here, but that the leda-free-6.3 is potentially buggy w.r.t. parsing rationals. To confirm, I'm afraid you'd have to ask at Algorithmic Solutions.

@lrineau Maybe you can also check the situation with 6.3 on Fedora? The Arch Linux LEDA package is constructed from http://www.algorithmic-solutions.info/free/Packages/LEDA-6.3-free-fedora-core-8-64-g++-4.1.2-mt.tar.gz (SHA256: 281057aa2077972d46e5a8e42cecb5633fdf7682bc4ff8ca83e31bde6ae06db9)...

**Comment by lrineau** [comment]
I can reproduce the issue with Fedora 24 and LEDA 6.3.

Actually, now I remember that we have already reported that issue to Algorithmic Solution, long ago, and they have fixed it in LEDA 6.4 and LEDA-numbers 6.4.

I have deactivated the test in 4f012205a923871acca1985336899aec64a36857, to "fix" the test suite.
